org.adempiere.base/src/org/compiere/wf/MWFNextCondition.java

```diff
@@ -342,5 +364,40 @@ public MWFNextCondition markImmutable() {
 		makeImmutable();
 		return this;
 	}
+	
+	/**
+	 * Executes SQLStatement of Record and returns result as Object
+	 * @param Record_ID
+	 * @return SQL statement execution result
+	 * @throws Exception
+	 */
+	private Object getSqlResult(int Record_ID) throws Exception {
+		String sqlStatement = (String) get_Value("SQLStatement");
+		String tableName = getAD_Column().getAD_Table().getTableName();
+		String pkName = tableName + "_ID"; // primary key name
+
+		Object result = null;
+
+		String resultSql = String.format("SELECT (%s) FROM %s WHERE %s = %d", sqlStatement, tableName, pkName, Record_ID);

```

Comment:
this looks strange - I thought it was just to run the SQL statement straight there
I haven't tested but I think the construction maybe is strange and different to the other SQLStatement approach on the application.
Some examples maybe?

Reply:
This approach is similar to Virtual Columns. Code is in org.compiere.model.GridTable.createSelectSql() method.
It uses query from SQLStatement as a subquery in SELECT statement to give user an ability to refer to current table.

For example if you need to get some Name from C_BPartner table while WF setted to work in AD_User, you can type:
`SELECT Name FROM C_BPartner WHERE C_BPartner_ID = AD_User.C_BPartner_ID`

And it will be transformed to:
`SELECT **(SELECT Name FROM C_BPartner WHERE C_BPartner_ID = AD_User.C_BPartner_ID)** FROM AD_User WHERE AD_User_ID = 1000000`

Otherwise user couldn't refer to columns of table from WF in this query.